pkg/kubelet/kubelet.go

```diff
@@ -1159,6 +1192,7 @@ func (kl *Kubelet) SyncPods(allPods []*api.Pod, podSyncTypes map[types.UID]metri
 	//      though the pod was not considered terminated by the apiserver).
 	// These two conditions could be alleviated by checkpointing kubelet.
 	pods := kl.filterOutTerminatedPods(allPods)
+	pods = kl.filterOutPodsPastActiveDeadline(pods)
```

Comment:
@derekwaynecarr, I see your point. It seems like that the `handle*` functions would cause duplicated events. They should each filters out terminated pods before processing. I will upload a PR to fix that, so you can leave your PR unchanged.